drivers/video/hm0360.c

```diff
@@ -0,0 +1,885 @@
+	}
+
+	drv_data->cur_frmrate = frmival->numerator;
+
+	LOG_DBG("FrameRate selected: %d", frmival->numerator);
+	LOG_DBG("HIRES Selected: %d", highres);
+	LOG_DBG("OSC DIV: %d", osc_div);
+
+	return 0;
+}
+
+static int hm0360_get_frmival(const struct device *dev, struct video_frmival *frmival)
+{
+	struct hm0360_data *drv_data = dev->data;
+
+	frmival->numerator = drv_data->cur_frmrate;
+	frmival->denominator = 1;
+
+	return 0;
+}
+
+static int hm0360_init_controls(const struct device *dev)
+{
+	int ret;
+	struct hm0360_data *drv_data = dev->data;
+	struct hm0360_ctrls *ctrls = &drv_data->ctrls;
+
+	ret = video_init_ctrl(&ctrls->hflip, dev, VIDEO_CID_HFLIP,
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret) {
+		return ret;
+	}
+
+	return video_init_ctrl(&ctrls->vflip, dev, VIDEO_CID_VFLIP,
+			       (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+}
+
+static int hm0360_init(const struct device *dev)
+{
+	const struct hm0360_config *config = dev->config;
+	struct video_frmival frmival;
+	int ret;
+
+	struct video_format fmt = {
+		.pixelformat = fmts[0].pixelformat,
+		.width = fmts[0].width_min,
+		.height = fmts[0].height_min,
+	};
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		/* I2C device is not ready, return */
+		return -ENODEV;
+	}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
+	/* Power up camera module */
+	if (config->pwdn.port != NULL) {
+		if (!gpio_is_ready_dt(&config->pwdn)) {
+			return -ENODEV;
+		}
+		ret = gpio_pin_configure_dt(&config->pwdn, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not clear power down pin: %d", ret);
+			return ret;
+		}
+	}
+#endif
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	/* Reset camera module */
+	if (config->reset.port != NULL) {
+		if (!gpio_is_ready_dt(&config->reset)) {
+			return -ENODEV;
+		}
+		ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not set reset pin: %d", ret);
+			return ret;
+		}
+		/* Reset is active high, has 1ms settling time*/
+		k_msleep(1);
+		gpio_pin_set_dt(&config->reset, 0);
+		k_msleep(1);
+	}
+#endif
+
+	uint32_t pid = 0x0000;
+
+	/*  Read Product ID and check if PID OK  */
+	ret = video_read_cci_reg(&config->bus, HM0360_CCI_ID, &pid);
+	if (ret < 0) {
+		LOG_ERR("Could not read product ID: %d", ret);
+		return ret;
+	}
+
+	if (pid != hm0360_PROD_ID) {
+		LOG_ERR("Incorrect product ID: 0x%02X", pid);
+		return -ENODEV;
+	}
+	LOG_DBG("Product ID is: %x", pid);
+
+	/* Reset camera registers */
+	ret = video_write_cci_reg(&config->bus, HM0360_CCI_RESET, 0x01);
+	if (ret) {
+		LOG_ERR("Could not write reset: %d", ret);
+		return ret;
+	}
+	/* Delay for 1ms. */
+	k_msleep(1);
+
+	/* Write Default registers */
+	ret = video_write_cci_multiregs16(&config->bus, hm0360_default_regs,
+					  ARRAY_SIZE(hm0360_default_regs));
+	if (ret < 0) {
+		LOG_ERR("Failed to set default registers");
+		return ret;
+	}
+
+	hm0360_set_fmt(dev, &fmt);
```

Comment:
I think you might have been missing a `ret = ` at this line.

```suggestion
	ret = hm0360_set_fmt(dev, &fmt);
```